packages/ariakit-utils/src/__tests__/focus-test.ts

```diff
@@ -0,0 +1,827 @@
+import * as dom from "../dom";
+import {
+  disableFocus,
+  disableFocusIn,
+  ensureFocus,
+  focusIfNeeded,
+  getAllFocusable,
+  getAllFocusableIn,
+  getAllTabbable,
+  getAllTabbableIn,
+  getClosestFocusable,
+  getFirstFocusable,
+  getFirstFocusableIn,
+  getFirstTabbable,
+  getFirstTabbableIn,
+  getLastTabbable,
+  getLastTabbableIn,
+  getNextTabbable,
+  getNextTabbableIn,
+  getPreviousTabbable,
+  getPreviousTabbableIn,
+  hasFocus,
+  hasFocusWithin,
+  isFocusable,
+  isTabbable,
+  restoreFocusIn,
+} from "../focus";
+
+jest.mock("../dom");
+
+// Abstract back and share?
+function getById<T extends HTMLElement = HTMLElement>(id: string) {
+  return document.getElementById(id) as T;
+}
+
+const mockDom = dom as jest.Mocked<typeof dom>;
+
+test("isFocusable", () => {
+  const element = document.createElement("div");
+
+  // Mock matches and isVisible because those functions are tested elsewhere.
```

Comment:
I'm not sure this is a good approach because it's testing implementation details. If we stop using `dom`'s `matches` and `isVisible` inside the `isFocusable` function, this test will fail, even though the new implementation work.

In the `ariakit-test-utils` package, we have an internal `__mock-get-client-rects.ts` file that addresses the JSDOM lack of support:

https://github.com/ariakit/ariakit/blob/278cff90b700de972ceae5114acc3b142ba0377d/packages/ariakit-test-utils/src/__mock-get-client-rects.ts#L1-L20

We should do something like that by default for `dom` and `focus` tests and avoid mocking internal dependencies. It's okay to duplicate that file inside this `__tests__` folder and import it into the test files.

Reply:
Would you prefer it is copied in or referenced from the other package?